include/number_types.hpp

```diff
@@ -0,0 +1,139 @@
+#pragma once
+
+#include "chainblocks.h"
+#include "common_types.hpp"
+#include <vector>
+#include <map>
+#include <cassert>
+
+enum class NumberType {
+  Float32,
+  Float64,
+  UInt8,
+  Int8,
+  Int16,
+  Int32,
+  Int64,
+  Invalid
```

Comment:
I would put `Invalid` as the first entry, since `0` is the default value for an enum. Easier to spot mistakes.